### PR TITLE
Remove test helpers (major changes #5)

### PR DESCRIPTION
**Pull request by FK78**
Resolves #3264 

**Overall change:** _Removed test helper imports_

**Code changes:**

- _ATIAnalytics import replaced._
- _App index test rewritten._
- _Added `@testing-library/jest-dom` to dev dependencies._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/App/index.test.jsx [hunk 1]
```diff
@@ -1,38 +1,78 @@
...
+    );
+    expect(BrowserRouter).toHaveBeenCalledWith(
+      {
+        children: expect.anything(),
+        data: 'someData!',
+        routes: ['someRoute'],
+      },
+      {},
+    );
+    const { getByText } = render(
```
Comment (jroebu14) [inline src/app/containers/App/index.test.jsx]: this doesn't prove what I think you're trying to prove

**Review by oluoluoxenfree — APPROVED** [review]

File: src/app/containers/App/index.test.jsx [hunk 2]
```diff
@@ -1,38 +1,78 @@
...
+BrowserRouter.mockImplementation(({ children }) => <div>{children}</div>);
+StaticRouter.mockImplementation(({ children }) => <div>{children}</div>);
 
 describe('ClientApp', () => {
-  shouldShallowMatchSnapshot(
-    'should render correctly',
-    <ClientApp data="someData!" routes={['someRoute']} />,
-  );
+  it('should render correctly', () => {
+    render(<ClientApp data="someData!" routes={['someRoute']} />);
```
Comment (jroebu14) [inline src/app/containers/App/index.test.jsx]: should be

```jsx
const { getByText } = render(<ClientApp data="someData!" routes={['someRoute']} />);
```

File: src/app/containers/App/index.test.jsx (hunk 1 above)
Comment (jroebu14) [reply src/app/containers/App/index.test.jsx]: see the comment on line 19

File: src/app/containers/App/index.test.jsx [hunk 3]
```diff
@@ -1,38 +1,70 @@
...
 import { ClientApp, ServerApp } from '.';
+import * as App from './App';
+
+jest.mock('./App', () => jest.fn(() => <div>I am the App component</div>));
+
+jest.mock('react-router-dom', () => ({
+  BrowserRouter: jest.fn(),
+  StaticRouter: jest.fn(),
+}));
+BrowserRouter.mockImplementation(({ children }) => <div>{children}</div>);
```
Comment (jroebu14) [inline src/app/containers/App/index.test.jsx]: Do these things need to be mocked for the tests to work?

Comment (ghost) [reply src/app/containers/App/index.test.jsx]: I think the main purpose of mocking them is make it possible to `assert` that they were called with the correct props.

Comment (jroebu14) [reply src/app/containers/App/index.test.jsx]: OK cool. I'd separate out the assertion on line 34 into it's own test and give the tests better titles e.g. `App is called with correct props` and `ClientApp renders the App component` but I'd make sure the Browser router mock is removed before testing the latter. Then do the same for ServerApp

**Review by jroebu14 — COMMENTED** [review]
LGTM but I'll leave the CO approval to @dr3 since he's more familiar with this stuff

**Review by dr3 — APPROVED** [review]

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**